external_data/data.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+from typing import List
+
+import requests
+from datetime import datetime
+import pytz
+from external_data.models.fund import Fund
+from external_data.models.round import Round
+from config import FLASK_ROOT
+from config import FUND_STORE_API_HOST
+from config import ROUND_STORE_API_HOST
+
+# Fund Store Endpoints
+FUNDS_ENDPOINT = "/funds/"
+FUND_ENDPOINT = "/funds/{fund_id}"
+
+# Round Store Endpoints
+ROUNDS_ENDPOINT = "/fund/{fund_id}"
+ROUND_ENDPOINT = "/fund/{fund_id}/round/{round_id}"
+
+
+def get_data(endpoint: str):
+    if endpoint[:8] == "https://":
+        response = requests.get(endpoint)
+        if response.status_code == 200:
+            data = response.json()
+        else:
+            return None
+    else:
+        data = get_local_data(endpoint)
+    return data
+
+
+def get_local_data(endpoint: str):
+    api_data_json = os.path.join(
+        FLASK_ROOT, "tests", "api_data", "endpoint_data.json"
+    )
+    fp = open(api_data_json)
+    api_data = json.load(fp)
+    fp.close()
+    if endpoint in api_data:
+        return api_data.get(endpoint)
+
+
+def get_funds() -> List[Fund] | None:
+    endpoint = FUND_STORE_API_HOST + FUNDS_ENDPOINT
+    response = get_data(endpoint)
+    if response and len(response) > 0:
+        funds = []
+        for fund in response:
+            funds.append(Fund.from_json(fund))
+        return funds
+    return None
+
+
+def get_fund(fund_id: str) -> Fund | None:
+    endpoint = FUND_STORE_API_HOST + FUND_ENDPOINT.format(fund_id=fund_id)
+    response = get_data(endpoint)
+    if response and "fund_id" in response:
+        fund = Fund.from_json(response)
+        if "rounds" in response and len(response["rounds"]) > 0:
+            for fund_round in response["rounds"]:
+                fund.add_round(Round.from_json(fund_round))
+        return fund
+    return None
+
+
+def get_rounds(fund_id: str) -> Fund | List:
+    endpoint = ROUND_STORE_API_HOST + ROUNDS_ENDPOINT.format(fund_id=fund_id)
+    response = get_data(endpoint)
+    print(endpoint)
+    rounds = []
+    if response and len(response) > 0:
+        for round_data in response:
+            rounds.append(Round.from_json(round_data))
+    return rounds
+
+
+def get_round(fund_id: str, round_id: str = None, date_submitted: datetime = None) -> Round | None:
+    """
+    Gets round from round store api using round_id if given.
+    If no round_id is provided, attempts to find the correct round
+    based on date_submitted time and open rounds for the given fund_id
+    """
+    fund_round = None
+    if round_id:
+        round_endpoint = ROUND_STORE_API_HOST + ROUND_ENDPOINT.format(
+            fund_id=fund_id, round_id=round_id
+        )
+        round_response = get_data(round_endpoint)
+        if round_response and "round_id" in round_response:
+            fund_round = Round.from_json(round_response)
+        if not fund_round:
```

Comment:
Ok, lets have this debate again...I don't like that it's not explicit that a comparison to None is being done here. If you disagree that's fine but I would prefer 
```python
if fund_round is not None:
```

Reply:
Well I'd argue that two wrongs don't make a right